pkg/controller/common/common.go

```diff
@@ -0,0 +1,70 @@
+package common
```

Comment:
I think packages named `common`, `utils`, etc, are very vague as they imply shared logic without clear separation of concerns. At the same time, it seems that the main entity of this file is the `CommonReconciler`, but it is not very clear if we are reconciling anything here. We are also constructing and returning heterogeneous things here e.g. `backup.PBM`, `mongo.Client`. scheme, k8s client. 

Maybe the scheme and the k8s client should remain in `ReconcilePerconaServerMongoDB` and add functions with receivers there for these. Then PBM should be organized separately and MongoClientProvider separately. 

For example,